src/TBuffer.cpp

```diff
@@ -1784,7 +1804,214 @@ void TBuffer::translateToPlainText( std::string & s )
             }
             continue;
         }
-        mMudLine.append( ch );
+
+        // PLACEMARKER: Incoming text decoding
+        // Used to double up the TChars for Utf-8 byte sequences that produce
+        // a surrogate pair (non-BMP):
+        bool isTwoTCharsNeeded = false;
+        if      (mpHost->mTelnet.getEncoding() == QLatin1String("ISO 8859-1")) {
+            mMudLine.append(QString(QChar::fromLatin1(ch)));
+        }
+        else if (mpHost->mTelnet.getEncoding() == QLatin1String("ISO 8859-2")) {
+            mMudLine.append(QString(decodeByteToIso_8859_2(ch)));
+        }
+        else if (mpHost->mTelnet.getEncoding() == QLatin1String("ISO 8859-3")) {
+            mMudLine.append(QString(decodeByteToIso_8859_3(ch)));
+        }
+        else if (mpHost->mTelnet.getEncoding() == QLatin1String("ISO 8859-4")) {
+            mMudLine.append(QString(decodeByteToIso_8859_4(ch)));
+        }
+        else if (mpHost->mTelnet.getEncoding() == QLatin1String("ISO 8859-10")) {
+            mMudLine.append(QString(decodeByteToIso_8859_10(ch)));
+        }
+        else if (mpHost->mTelnet.getEncoding() == QLatin1String("ISO 8859-15")) {
+            mMudLine.append(QString(decodeByteToIso_8859_10(ch)));
+        }
+        else if (mpHost->mTelnet.getEncoding() == QLatin1String("ISO 8859-16")) {
+            mMudLine.append(QString(decodeByteToIso_8859_16(ch)));
+        }
+        else if (mpHost->mTelnet.getEncoding() == QLatin1String("WINDOWS-1250")) {
+            mMudLine.append(QString(decodeByteToWindows_1250(ch)));
+        }
+        else if (mpHost->mTelnet.getEncoding() == QLatin1String("WINDOWS-1251")) {
+            mMudLine.append(QString(decodeByteToWindows_1250(ch)));
+        }
+        else if (mpHost->mTelnet.getEncoding() == QLatin1String("WINDOWS-1252")) {
+            mMudLine.append(QString(decodeByteToWindows_1252(ch)));
+        }
+        else if (mpHost->mTelnet.getEncoding() == QLatin1String("UTF-8")) {
```

Comment:
This is over my head and I won't catch up on this in time next release so leaving to @ahmedcharles to verify.